File: greedy/greedy3.py
inputs = input()

count0 = 0
count1 = 0
if inputs[0] == "1":
    count0 += 1
else:
    count1 += 1

for i in range(0, len(inputs) - 1):
    if inputs[i] != inputs[i + 1]:
        if inputs[i + 1] == "1":
            count0 += 1
        else:
            count1 += 1

print(min(count0, count1))

File: greedy/greedy3.java
package algorithm.greedy;

import java.util.Scanner;

public class greedy3 {

    public static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        String inputs = sc.nextLine();
        int count0=0;
        int count1=0;
        if(inputs.charAt(0)-'0'==0){
            count1++;
        }else{
            count0++;
        }
        for(int i=0; i<inputs.length()-1;i++){
            if(inputs.charAt(i) != inputs.charAt(i+1)){
                if(inputs.charAt(i+1)=='1'){
                    count0++;
                }else{
                    count1++;
                }
            }
        }
        System.out.println(Math.min(count0,count1));
    }
}
